### ISSUE TITLE
HiGHS cmake instructions result in bad linking

### ISSUE THREAD

**Issue by silky** [ERGO-Code/HiGHS#1676]
If you naively follow the README instructions for building HiGHS locally, you end up with a badly linked executable in `/usr/local/bin`.

To reproduce:

```shell
> docker run -it ubuntu:latest
root@...:/# apt-get -y update && apt-get -y install build-essential git cmake protobuf-compiler
root@...:/# git clone https://github.com/ERGO-Code/HiGHS.git
root@...:/# cd HiGHS
root@...:/HiGHS# cmake -S . -B build
root@...:/HiGHS# cmake --build build
root@...:/HiGHS# cmake --install build
root@...:/HiGHS# highs
highs: error while loading shared libraries: libhighs.so.1: cannot open shared object file: No such file or directory
```

Note that the exe in `./build/bin/highs` works fine:

```shell
root@....:/HiGHS# ./build/bin/highs 
Please specify filename in .mps|.lp|.ems format.
```

So I think it's just a problem with connecting up the `LD_LIBRARY_PATH` or something; but probably a cmake expert will know the answer :)

**Comment by Coloquinte** [comment]
It's not cmake but Linux/docker.
You need to run `ldconfig` after installing, so the list of shared libraries is updated.
[Cmake doesn't do it for us](https://cmake.org/pipermail/cmake/2016-June/063713.html)

**Comment by metab0t** [comment]
You can change the `RPATH` for the highs executable like https://github.com/metab0t/highsbox/blob/edd84ac1767b9df82fba2cb635ea34ef87d95ab9/setup.py#L93

Ensure `libhighs.so` is in the `<your_highs_executable_path>/../lib` folder.

Run the following commands on Linux:
```
patchelf --set-rpath '$ORIGIN/../lib' <your_highs_executable_path>
```

On Mac, run:
```
install_name_tool -add_rpath '@loader_path/../lib' <your_highs_executable_path>
```

**Comment by galabovaa** [comment]
I have made an attempt to address this issue in branch rpath-bin:

https://github.com/ERGO-Code/HiGHS/pull/1745/files

Would you please check whether this removes both 
> the need to modify paths on linux @silky 
> the need for your patch, @metab0t

**Comment by silky** [comment]
@galabovaa naively following my initial instructions above, I now get

```
CMake Error at CMakeLists.txt:191 (cmake_policy):
  Policy "CMP0138" is not known to this version of CMake.
```

on that branch :(

**Comment by galabovaa** [comment]
Thank you! I will edit cmake accordingly

**Comment by galabovaa** [comment]
@silky this fix is included in the latest release v1.7.1. Closing this issue, please re-open it, or a new one, if you encounter any further issues.
